src/box/applier.cc

```diff
@@ -1363,6 +1384,7 @@ applier_thread_signal_ack(struct cmsg *base)
 	fiber_cond_signal(&applier->thread.writer_cond);
 	applier->thread.has_acks_to_send = true;
 	applier->thread.txn_last_tm = msg->txn_last_tm;
+	applier->thread.last_sync = msg->sync;
```

Comment:
You're correct, but this is fine.

Let's for simplicity call the moment, when a linearizable request is started, a `start`.
Let's call the master a node performing a linearizable transaction. So it issues a request with some `sync` and waits for vclocks in response with the same sync.

Having a `sync` in request and response guarantees, that an ACK coming with that `sync` was surely sent **after** `start`. This is enough. This means that `ack_vclock` coming together with that `sync` in an ACK **was** the greatest vclock replica has ever reported to anyone  at the moment of `start`.

Now, when we have `replication_connect_quorum = Q`, master receives `N - Q + 1` acks with sync.
At least one of `ack_vclock`s contains everything that **was** committed at the moment of `start`. Because at least one of `N - Q + 1` nodes must have ACKed the committed transaction.

To be more precise, we need componentwise max of `ack_vclock`s and name it `max_vclock`.

This all, in turn, means, that even if some other master has committed something **before** `start`, this something is surely reflected in `max_vclock`. And once we wait for this vclock, we have reached the linearization point.

P.S. of course, at the moment of responding to master's request, replica might already be writing something new. But this something is guaranteed to commit later, than `start`. So we don't care about seeing it.

Reply:
Thanks for the explanation .Sounds edgy, but ok.